video_xblock/backends/base.py

```diff
@@ -0,0 +1,93 @@
+import abc
+import pkg_resources
+import re
+
+from HTMLParser import HTMLParser
+from webob import Response
+from xblock.fragment import Fragment
+from xblock.plugin import Plugin
+
+from django.template import Template, Context
+
+
+html_parser = HTMLParser()
+
+
+class BaseVideoPlayer(Plugin):
+    """
+    Inherit your video player class from this class
+    """
+    __metaclass__ = abc.ABCMeta
+
+    entry_point = 'video_xblock.v1'
+
+    @abc.abstractproperty
+    def url_re(self):
+        """
+        Regex (list) to match video url
+        """
+        return [] or re.compile('') or ''
+
+    @abc.abstractmethod
+    def get_frag(self, **context):
+        """
+        Returns a Fragment required to render video player on the client side.
+        """
+        return Fragment('<video />')
+
+    @abc.abstractmethod
+    def media_id(self, href):
+        """
+        Extracts Platform's media id from the video url.
+        E.g. https://example.wistia.com/medias/12345abcde -> 12345abcde
+        """
+
+        return ''
+
+    def get_player_html(self, **context):
+        """
+        Renders self.get_frag as a html string and returns it as a Response.
+        This method is used by VideoXBlock.render_player()
+
+        Rendering sequence is set to JS must be in the head tag and executed
+        before initializing video components.
+        """
+        frag = self.get_frag(**context)
+        html = Template(self.resource_string("../static/html/base.html"))
+        return Response(
+            html_parser.unescape(
+                html.render(
+                    Context({
+                        'head': frag.head_html(),
+                        'foot': frag.foot_html(),
+                        'body': frag.body_html()
+                    })
+                )
+            ),
+            content_type='text/html'
+        )
+
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    def render_resource(self, path, **context):
+        """
+        Renders static resource using provided context
+
+        Returns: django.utils.safestring.SafeText
```

Comment:
Generally, XBlocks should not be very django-dependant, @polesye  what you think? Is it ok?